src/java.security.jgss/share/native/libj2gss/NativeUtil.c

```diff
@@ -455,6 +455,14 @@ void throwOutOfMemoryError(JNIEnv *env, const char *message) {
     throwByName(env, "java/lang/OutOfMemoryError", message);
 }
 
+static jsize
+safe_jsize(size_t n)
+{
+    jsize res = (jsize)n;
```

Comment:
I'm not convinced this function is safe. I suspect it invokes undefined behavior more than once.